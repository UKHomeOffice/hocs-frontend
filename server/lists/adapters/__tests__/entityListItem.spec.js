const entityListItemsAdapter = require('../entityListItems');

const mockLogger = {
    debug: () => { },
    info: () => { },
    warn: () => { },
    error: () => { }
};

describe('Entity List Items Adapter', () => {

    it('should transform data', async () => {
        const mockData = [
            { simpleName: 'First', data: { title: 'Title First' }, active: true },
            { simpleName: 'Second', data: { title: 'Title Second' }, active: true },
            { simpleName: 'Third', data: { title: 'Title Third' }, active: false },
        ];

        const results = await entityListItemsAdapter(mockData, { logger: mockLogger });

        expect(results).toBeDefined();
        expect(results).toMatchSnapshot();
    });
});
