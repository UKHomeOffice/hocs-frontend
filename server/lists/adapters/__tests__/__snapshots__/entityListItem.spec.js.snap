// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entity List Items Adapter should transform data 1`] = `
Array [
  Object {
    "label": "Title First",
    "value": "First",
  },
  Object {
    "label": "Title Second",
    "value": "Second",
  },
  Object {
    "label": "Title Third",
    "value": "Third",
  },
]
`;
